mvrx-compose/src/main/kotlin/com/airbnb/mvrx/compose/MavericksComposeExtensions.kt

```diff
@@ -61,16 +64,27 @@ inline fun <reified VM : MavericksViewModel<S>, reified S : MavericksState> mave
     val savedStateRegistryOwner = scope as? SavedStateRegistryOwner ?: error("LifecycleOwner must be a SavedStateRegistryOwner!")
     val savedStateRegistry = savedStateRegistryOwner.savedStateRegistry
     val viewModelClass = VM::class
+
+    val localView = LocalView.current
     val viewModelContext = remember(scope, activity, viewModelStoreOwner, savedStateRegistry) {
-        when (scope) {
-            is Fragment -> {
-                val args = scope.arguments?.get(Mavericks.KEY_ARG)
-                FragmentViewModelContext(activity, args, scope)
-            }
-            else -> {
-                val args = activity.intent.extras?.get(Mavericks.KEY_ARG)
-                ActivityViewModelContext(activity, args, viewModelStoreOwner, savedStateRegistry)
+        var view: View? = localView
+        var fragment: Fragment? = null
```

Comment:
I think the try/catch approach is fine, and is maybe preferable to not rely on the implementation detail of the `fragment_container_view_tag` tag. Main thing I would say though is to extract this to a dedicated function and add some documentation and where this pattern was taken from (eg point to the androidx code)

Reply:
The issue here is that this whole factory is inline. So any function we want to extract have to be public in na namespace of mavericks

Reply:
I solved that by using `@InternalMavericksApi`, I hope that's enough

Reply:
@jakoss that's fine. in the future for inline functions you can also make them `internal` and annotate with `@PublishedApi`